scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+CLONE INSTANCE FROM '$WSREP_SST_OPT_REMOTE_JOINER_USER'@'$LHOST':$WSREP_SST_OPT_LPORT IDENTIFIED BY '$WSREP_SST_OPT_REMOTE_JOINER_PSWD' $REQUIRE_SSL;
+EOF
+
+        wsrep_log_debug "JOINER CLONE ACTION SQL: $CLONE_EXECUTE_SQL $MYSQL_PWD $MYSQL_RCLIENT"
+        CLONE_EXECUTE=`cat $CLONE_EXECUTE_SQL` || :
+#        wsrep_log_debug "-> $CLONE_EXECUTE"
+        
+        # Actual cloning process
+        wsrep_log_info "JOINER CLONE ACTION: cloning"
+        LOCALOUTPUT=`$MYSQL_RCLIENT --connect-timeout=60 $CLIENT_SSL_OPTIONS < $CLONE_EXECUTE_SQL  2>&1 || RC=$?` 
+        wsrep_log_info "JOINER CLONE ACTION: cloning done $RC"
+        wsrep_log_debug "-> LOCALOUTPUT: $LOCALOUTPUT"
+
+
+        # We force the signal to be 0 because we KNOW that with clone when using the mysqld directly (not mysqld_safe) the daemon is shutdown at the end of the clone process.
+        # However an error is returned because server cannot be restarted 
+        # This error does not indicate a cloning failure. It means that the recipient MySQL server instance must be started again manually after the data is cloned. 
+        if [ "$RC" -ne 0 ]; then
+            RC=0
+            wsrep_log_debug " JOINER CLONE ACTION abruptly terminated, but we can continue"
+        fi
+
+        # If still there we will manually shutdown
+        LOCALOUTPUT=`$MYSQL_RCLIENT -e "SHUTDOWN"  2>&1 || :` 
+        wsrep_log_debug "-> LOCALOUTPUT: $LOCALOUTPUT"
+
+        if [ "$RC" -ne 0 ]
+        then
+            wsrep_log_error "Clone command returned code $RC"
+            wsrep_log_debug "JOINER RETURN ERROR $MYSQL_RCLIENT"
+            cat $CLONE_EXECUTE_SQL >> /dev/stderr
+            # Attempt to shutdown recipient daemon
+            eval $MYSQL_RCLIENT -e "SHUTDOWN"
+            wsrep_log_info "Recipient shutdown: $?"
+            case $RC in
+            *)  RC=255 # unknown error
+                ;;
+            esac
+            exit $RC
+        fi
+        CLEANUP_CLONE_SSL="yes"
+        cleanup_donor
+    else # BYPASS
+        wsrep_log_info "Bypassing state dump."
+
+        # Instruct recipient to shutdown
+        # export MYSQL_PWD="$WSREP_SST_OPT_REMOTE_JOINER_PSWD"
+        wsrep_log_info "BYPASS SENDING IST_FILE TO JOINER NetCat: nc -w 1 $SST_HOST_STRIPPED $WSREP_SST_OPT_REMOTE_HOSTPORT"
+        echo "$WSREP_SST_OPT_GTID<EOF>" | nc -w 1 $SST_HOST_STRIPPED $WSREP_SST_OPT_REMOTE_HOSTPORT || :
+        sleep 2
+        wsrep_log_debug "-> Exiting with gtid: $WSREP_SST_OPT_GTID"
+    fi
+
+    wsrep_log_debug "-> SENDING DONE_MESSAGE $WSREP_SST_OPT_GTID"
+    # DONOR must be clean BEFORE exit given user is removed on DONE
+
+    echo "done $WSREP_SST_OPT_GTID"
+
+elif [ "$WSREP_SST_OPT_ROLE" = "joiner" ]
+then
+
+    JOINER_CLONE_HOST=""
+    JOINER_CLONE_PORT=""
+
+    CLEANUP_FILES=""
+
+    trap sig_cleanup_joiner HUP PIPE INT TERM
+    trap cleanup_joiner EXIT
+
+    wsrep_check_programs grep
+    wsrep_check_programs ps
+    wsrep_check_programs find
```

Comment:
Script does not exit if any of the command is not found

Reply:
I think it exists. The script runs without set +e, so it exists any time the function returns non-zero

Reply:
yup, there is `set -o errexit` at the top of the script